cmake/modules/configuration_files.cmake

```diff
@@ -116,4 +116,4 @@ DTC_OVERLAY_FILE=\"dts1.overlay dts2.overlay\"")
 # The DTC_OVERLAY_FILE variable is now set to its final value.
 zephyr_boilerplate_watch(DTC_OVERLAY_FILE)
 
-zephyr_get(OVERLAY_CONFIG SYSBUILD LOCAL)
+zephyr_get(EXTRA_CONF_FILE SYSBUILD LOCAL VAR EXTRA_CONF_FILE OVERLAY_CONFIG)
```

Comment:
This seems to be a "silent" deprecation; will there be a way to warn users about this at some point? (Maybe it's later in the PR, but the commit message says that this commit is doing the deprecation of OVERLAY_CONFIG.)